modules/mono/mono_gd/gd_mono.cpp

```diff
@@ -501,26 +501,51 @@ bool GDMono::_load_project_assembly() {
 #endif
 
 #ifdef GD_MONO_HOT_RELOAD
+void GDMono::reload_failure() {
+	if (++project_load_failure_count >= 5) {
+		// After reloading a project has failed 5 times in a row, update the path and modification time
+		// to stop any further attempts at loading this assembly, which probably is never going to work anyways
+		project_load_failure_count = 0;
+
+		ERR_PRINT_ED(".NET: Giving up on assembly reloading. Please restart the editor if unloading was failing.");
+
+		String assembly_name = GLOBAL_GET("dotnet/project/assembly_name");
+		if (assembly_name.is_empty()) {
+			assembly_name = ProjectSettings::get_singleton()->get_safe_project_name();
+		}
```

Comment:
Should be able to use `path::get_csharp_project_name` here.

```suggestion
		String assembly_name = path::get_csharp_project_name();
```